tools/log-collector/log_collector.go

```diff
@@ -748,3 +734,66 @@ func (l *LogCollector) getResourceObjectsWithOwnerRef(resourcePath string,
 	}
 	return objects
 }
+
+func (l *LogCollector) getClusterNodes() {
+
+	var nodeList unstructured.UnstructuredList
+	gvk := schema.GroupVersionKind{Kind: NodeKind, Version: CoreGv}
+	nodeList.SetGroupVersionKind(gvk)
+	err := l.k8sClient.List(context.Background(), &nodeList)
+	if err != nil {
+		log.Error("Unable to Fetch Node List")
+	}
+	for _, node := range nodeList.Items {
+		resourceDir := filepath.Join(node.GetKind())
+		yErr := l.writeYaml(resourceDir, node)
+		if yErr != nil {
+			log.Error("Unable to Write Node Yaml")
+		}
+	}
+}
+
+// WriteNs writes yaml of requested namespaces
+func (l *LogCollector) WriteNs(namespaceObjects unstructured.UnstructuredList) error {
+	for _, obj := range namespaceObjects.Items {
+		resourceDir := filepath.Join(obj.GetKind())
+		if !l.Clustered {
+			if contains(l.Namespaces, obj.GetName()) {
+				eLrr := l.writeYaml(resourceDir, obj)
+				if eLrr != nil {
+					return eLrr
+				}
+			}
+		} else {
+			eLrr := l.writeYaml(resourceDir, obj)
+			if eLrr != nil {
+				return eLrr
+			}
+		}
+	}
+	return nil
+}
+
+// checkNamespaces taken all given ns from user and checks the same in cluster and write it YAML's
+func (l *LogCollector) checkNamespaces() error {
+	log.Info("Checking Namespaces")
+	coreGV, cgErr := l.getAPIGVResources(CoreGv)
+	if cgErr != nil {
+		return cgErr
+	}
+	namespaceResource := getResourceByName(coreGV, Namespaces)
+	namespaceObjects := l.getResourceObjects(getAPIGroupVersionResourcePath(CoreGv), &namespaceResource)
+	allNamespaces := getObjectsNames(namespaceObjects)
+
+	if len(l.Namespaces) != 0 && !l.isSubset(allNamespaces) {
```

Comment:
optimise func `isSubset` ... use `sets` data type and it's native funcs

Reply:
resolved..made changes from slice to map and reduced the loop. in our cases there are multiple namespaces to be selected from a array of namespaces